sn_node/src/put_validation.rs

```diff
@@ -438,35 +438,41 @@ impl Node {
                     Err(ProtocolError::FailedToDecypherTransfer) => continue,
                     // transfer invalid
                     Err(e) => return Err(e),
-                    // transfer ok
-                    Ok(cns) => cash_notes = cns,
+                    // transfer ok, add to cash_notes and continue as more transfers might be ours
+                    Ok(cns) => cash_notes.extend(cns),
                 },
-                Transfer::Encrypted(_) => continue, // we already have our cash notes, so we can skip this one
```

Comment:
There might be multiple Transfers for us! So we can't stop at the first one!

Reply:
But the multiple are all picked out by `verify_and_unpack_transfer` no?

Reply:
This unpacks only one of the multiple transfers! Each transfer might contain multiple CashNotes. We want all the cashnotes from all the transfers that are ours!